src/v/test_utils/async.h

```diff
@@ -50,4 +51,32 @@ requires ss::ApplyReturns<Predicate, bool> ||
           });
       }));
 }
+
+// When a test expects that any background fibers should complete promptly,
+// and wants to send a barrier through all the inter-CPU queues to ensure
+// that earlier-submitted tasks have reached their destination cores already.
+//
+// This is useful in tests that know they have put the system into a state where
+// it will get to a known state once all the non-i/o-blocking tasks in flight
+// have completed, such as background release of quota units.
+//
+// **Be aware** that there are assumptions to using this:
+// A) That your test code is using the same default scheduling group that
+//    this routine will run within.
+// B) That debug mode scheduling randomization in seastar does not re-order
+//    things so dramatically that our messages can go between every core and
+//    back before a future that was already ready on some shard gets run.
+inline void flush_tasks() {
+    // Ensure anything in inter-CPU queues before we entered the function
+    // has drained: this is an all-to-all to cover the full mesh of queues
+    // between cores.
+    ss::smp::invoke_on_all([]() {
+        return ss::smp::invoke_on_all([]() { return ss::yield(); });
+    }).get();
```

Comment:
It might be worth noting in the comment that this only suitable to call on a seastar thread, not a vanilla fiber due to `get()`.